bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/thing/ThingResource.java

```diff
@@ -411,6 +411,7 @@ public Response updateConfiguration(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) St
         } catch (Exception ex) {
             logger.error("Exception during HTTP PUT request for update config at '{}' for thing '{}': {}",
                     uriInfo.getPath(), thingUID, ex.getMessage());
+            logger.debug("Exception during HTTP PUT request ", ex.getMessage());
```

Comment:
This doesn't actually add any additional value, I think you meant something like this:

```java
if (logger.isDebugEnabled()) {
    logger.debug("", ex);
}
```
right?

Reply:
isn´t the `logger.isDebugEnabled()` call only required when extensive String concats or other  "expensive" operations are done before the call to `logger.debug` actually happens? Doing only `logger.debug("oops", ex)` should be okay here.

Reply:
Sure, good point - yes it is!

What I actually _meant_ to write down (should have copy-and-pasted it) was:

```java
if (logger.isDebugEnabled()) {
    logger.error("", ex);
}
```

but simply a 

```java
logger.debug("", ex);
```

would totally suffice of course.

Reply:
@htreu Do you consider to change `ex.getMessage()` to `ex`?

Reply:
Yes, on a second though I also wouldn't mind to have the full exception with stack trace in the original logging. Since we already log an error we consider this as a critical bug which results typically from coding mistake.

@cdjackson wdyt?

Reply:
@cdjackson ping!

Reply:
Sorry - not sure why I didn't see any of these comments (I don't seem to have any emails). Anyway, yes, I would be perfectly happy if we just had the single log and it logged the full stack track.

Do you want me to up update this PR with such implementation? (might be early next week as I'm now in DE/FR enroute to the conference).

Reply:
I would also vote for logging the full exception `ex` instead of just `ex.getMessage()`.